### ISSUE TITLE
Track - Workspace list doesn't show up when categorizing track expense

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#53274]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.68-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** No
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes, reproducible on both
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://github.com/Expensify/App/pull/52737
**Email or phone of affected tester (no customers):** nathanmulugetatesting+2212@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
Pre condition: Have multiple workspaces
1. Navigate to staging.new.expensify.com
2. Navigate to self DM 
3. Create a track expense
4. Click on Categorize it
5. Select a category 
6. Go back one step

## Expected Result:
When categorizing it workspace list should show up for the user to select from

## Actual Result:
No workspace list shows up on step 4 but category list shows up, then on step 6 workspace list shows up when going back

## Workaround:
Unknown

## Platforms:
- [x] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/a0750bce-5382-40a4-81b3-d77a3c748758

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by nkdengineer** [comment]
It's expected here https://github.com/Expensify/App/issues/52253

**Comment by mountiny** [comment]
Yea based on the instructions from the issue this is expected cc @trjExpensify  can you please double check and close it so? Thanks

**Comment by trjExpensify** [comment]
Yes, no workspace list anymore. But @nkdengineer, going back shouldn't then show the list of workspaces. I think we should address that.

**Comment by mountiny** [comment]
Assigned the C and C+ from the PR that changed it, @nkdengineer could you please follow up to implement the behaviour that Tom mentioned?

**Comment by nkdengineer** [comment]
@Ollyws The PR is ready for review.

@trjExpensify Please help to check the result in the [PR](https://github.com/Expensify/App/pull/53420) and confirm if it's the correct behavior.

**Comment by jliexpensify** [comment]
Looks like Kadie is now OOO so I can take over. 

@Ollyws - do we need a checklist?

**Comment by jliexpensify** [comment]
Payment Summary:

- C: @nkdengineer $250 (to be paid via Upworks)
- C+: @Ollyws $250 (to be paid via ND requests)

**Comment by jliexpensify** [comment]
Sorry before I do anything - can someone (maybe @mountiny) confirm that this needs payment? I assume yes, since @nkdengineer implemented some new behaviours.

**Comment by nkdengineer** [comment]
@trjExpensify Can you confirm this [comment](https://github.com/Expensify/App/issues/53274#issuecomment-2564971404)? cc @mountiny

**Comment by jliexpensify** [comment]
@mountiny Hi, sorry just bumping again - does this require payment, or is it part of a larger job (and therefore will be paid out in that job)?

**Comment by jliexpensify** [comment]
Thanks @mountiny!
